core-web/libs/sdk/analytics/src/lib/plugin/dot-analytics.plugin.ts

```diff
@@ -1,37 +1,44 @@
-import { ANALYTICS_PAGEVIEW_EVENT } from '../shared/analytics.constants';
-import { sendAnalyticsEventToServer } from '../shared/analytics.http';
-import { DotAnalyticsConfig, EventType, PageViewEvent } from '../shared/analytics.model';
-import { createAnalyticsPageViewData } from '../shared/analytics.utils';
+import { sendAnalyticsEventToServer } from '../shared/dot-content-analytics.http';
+import {
+    DotAnalyticsPayload,
+    DotContentAnalyticsConfig
+} from '../shared/dot-content-analytics.model';
 
 /**
- * The dotAnalytics plugin.
+ * Analytics plugin for tracking page views and custom events in DotCMS applications.
+ * This plugin handles sending analytics data to the DotCMS server, managing initialization,
+ * and processing both automatic and manual tracking events.
  *
- * @param {DotAnalyticsConfig} config - The analytics configuration.
- * @returns {Object} - The dotAnalytics plugin.
+ * @param {DotAnalyticsConfig} config - Configuration object containing API key, server URL,
+ *                                     debug mode and auto page view settings
+ * @returns {Object} Plugin object with methods for initialization and event tracking
  */
-export const dotAnalyticsPlugin = (config: DotAnalyticsConfig) => {
+export const dotAnalytics = (config: DotContentAnalyticsConfig) => {
     let isInitialized = false;
 
     return {
         name: 'dot-analytics',
         config,
 
-        initialize: ({ config }: { config: DotAnalyticsConfig }) => {
-            if (!config.server) {
-                throw new Error('DotAnalytics: Server URL is required');
-            }
-
+        /**
+         * Initialize the plugin
+         */
+        initialize: (params: {
+            config: DotContentAnalyticsConfig;
```

Comment:
Perhaps consider using [Zod](https://zod.dev/) to validate the configuration properly.

Reply:
I just will show a console message. No throw Error because breaks the execution. If the necessary attributes are not met, the plugins of dotAnalytic are not loaded.